java/src/org/openqa/selenium/bidi/LogInspector.java

```diff
@@ -0,0 +1,114 @@
+package org.openqa.selenium.bidi;
+
+import org.openqa.selenium.WebDriver;
+
+import org.openqa.selenium.bidi.log.BaseLogEntry;
+import org.openqa.selenium.bidi.log.ConsoleLogEntry;
+import org.openqa.selenium.bidi.log.GenericLogEntry;
+import org.openqa.selenium.bidi.log.JavascriptLogEntry;
+import org.openqa.selenium.bidi.log.Log;
+import org.openqa.selenium.bidi.log.LogEntry;
+import org.openqa.selenium.internal.Require;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class LogInspector implements AutoCloseable {
+
+  private final List<Consumer<ConsoleLogEntry>> consoleLogListeners = new LinkedList<>();
+  private final List<Consumer<JavascriptLogEntry>> jsExceptionLogListeners = new LinkedList<>();
+  private final List<Consumer<JavascriptLogEntry>> jsLogListeners = new LinkedList<>();
+  private final List<Consumer<GenericLogEntry>> genericLogListeners = new LinkedList<>();
+  private final Set<String> browsingContextIds;
+
+  private final BiDi bidi;
+
+  public LogInspector(WebDriver driver) {
+    this(new HashSet<>(), driver);
+  }
+
+  public LogInspector(String browsingContextId, WebDriver driver) {
```

Comment:
Maybe we could just do `this(Collections.singleton(browsingContextId), driver);`

Reply:
We can call the constructor but it does not allow me to do null checks in that case. Which need to happen before we call the constructor but when calling using `this(Collections.singleton(browsingContextId), driver);` it has to be the first statement.

Reply:
@pujagani - The null check for the contents of the collection would already be done by `Collections.singleton()` so we don't need to be doing it again. If we need it to be done explicitly we can create a private static method 

```java
private static Set<String> asSet(String text) {
    Require.nonNull("Browsing context id", text);
    return Collections.singleton(text);
}
```

and now within the constructor you can do 

```java
this(asSet(browsingContext), driver);
```

What do you think ?

Reply:
Ok. So `Collections.singleton()` doesn't do a null check.My bad. But my suggestion to do it via a static method should still do the trick and allow us to chain constructor calls.

Reply:
Thank you for your suggestion. The `Require.nonNull` does that already, it returns a value. I somehow missed that. Now we are able to chain constructors.